cirq/study/resolver.py

```diff
@@ -237,6 +237,16 @@ def _from_json_dict_(cls, param_dict, **kwargs):
         return cls(dict(param_dict))
 
 
+class ResolvableValue(Protocol):
+    @doc_private
+    def _resolver_value_(self) -> Any:
```

Comment:
Maybe we should call it `_resolved_value_` to match what's in the docstring below (I think that explanation is pretty clear). Also, I'd suggest moving this protocol into `resolve_parameters.py` alongside the other protocols that are related to parameter resolution, so we can document the whole process there.